tests/test_commands.py

```diff
@@ -95,7 +96,9 @@ def teardown():
                 # error when switching to the db 9 because we're not authenticated yet
                 # setting the password on the connection itself triggers the
                 # authentication in the connection's `on_connect` method
-                r.connection.password = temp_pass
+                r.connection.credential_provider = StaticCredentialProvider(
+                    password=temp_pass
+                )
```

Comment:
This... shouldn't be required since the static credential provider is now lazily created?